File: system-test/dotnet/SystemTests/Latest/AcceptanceTests/PlaceOrderPositiveIsolatedTest.cs
using SystemTests.Latest.AcceptanceTests.Base;
using Optivem.Testing;

namespace SystemTests.Latest.AcceptanceTests;

[Collection("Isolated")]
[Trait("Category", "isolated")]
public class PlaceOrderPositiveIsolatedTest : BaseAcceptanceTest
{
    [Theory]
    [ChannelData(ChannelType.UI, ChannelType.API)]
    public async Task ShouldRecordPlacementTimestamp(Channel channel)
    {
        (await Scenario(channel)
            .Given().Clock().WithTime("2026-01-15T10:30:00Z")
            .When().PlaceOrder()
            .Then().ShouldSucceed()
            .And().Clock())
            .HasTime("2026-01-15T10:30:00Z");
    }

    [Theory]
    [ChannelData(ChannelType.UI, ChannelType.API)]
    public async Task ShouldApplyFullPriceWithoutPromotion(Channel channel)
    {
        await Scenario(channel)
            .Given().Product().WithUnitPrice(20.00m)
            .And().Promotion().WithActive(false)
            .And().Country().WithTaxRate(0.00m)
            .When().PlaceOrder().WithQuantity(5)
            .Then().ShouldSucceed()
            .And().Order()
            .HasTotalPrice(100.00m);
    }

    [Theory]
    [ChannelData(ChannelType.UI, ChannelType.API)]
    public async Task ShouldApplyDiscountWhenPromotionIsActive(Channel channel)
    {
        await Scenario(channel)
            .Given().Product().WithUnitPrice(20.00m)
            .And().Promotion().WithActive(true).WithDiscount(0.5m)
            .And().Country().WithTaxRate(0.00m)
            .When().PlaceOrder().WithQuantity(5)
            .Then().ShouldSucceed()
            .And().Order()
            .HasTotalPrice(50.00m);
    }
}

File: system-test/java/src/test/java/com/optivem/shop/systemtest/legacy/mod10/acceptance/PlaceOrderPositiveIsolatedTest.java
package com.optivem.shop.systemtest.legacy.mod10.acceptance;

import com.optivem.shop.systemtest.legacy.mod10.acceptance.base.BaseAcceptanceTest;
import com.optivem.shop.testkit.channel.ChannelType;
import com.optivem.testing.Channel;
import com.optivem.testing.Isolated;

import org.junit.jupiter.api.TestTemplate;

@Isolated
class PlaceOrderPositiveIsolatedTest extends BaseAcceptanceTest {
    @TestTemplate
    @Channel({ChannelType.UI, ChannelType.API})
    void shouldApplyFullPriceOnWeekday() {
        scenario
                .given().product().withUnitPrice(20.00)
                .and().promotion().withActive(false)
                .and().country().withTaxRate("0.00")
                .and().clock().withWeekday()
                .when().placeOrder().withQuantity(5)
                .then().shouldSucceed()
                .and().order()
                    .hasTotalPrice(100.00);
    }

    @TestTemplate
    @Channel({ChannelType.UI, ChannelType.API})
    void shouldApplyDiscountWhenPromotionIsActive() {
        scenario
                .given().product().withUnitPrice(20.00)
                .and().promotion().withActive(true).withDiscount("0.5")
                .and().country().withTaxRate("0.00")
                .when().placeOrder().withQuantity(5)
                .then().shouldSucceed()
                .and().order()
                    .hasTotalPrice(50.00);
    }

    @TestTemplate
    @Channel({ChannelType.UI, ChannelType.API})
    void shouldRecordPlacementTimestamp() {
        scenario
                .given().clock()
                    .withTime("2026-01-15T10:30:00Z")
                .when().placeOrder()
                .then().shouldSucceed()
                .and().clock()
                    .hasTime("2026-01-15T10:30:00Z");
    }
}
